format/spec.md

```diff
@@ -1237,17 +1237,37 @@ Content file (data or delete) is serialized as a JSON object according to the fo
 | **`equality-ids`**       |`JSON list of int: Field ids used to determine row equality in equality delete files`|`[1]`|
 | **`sort-order-id`**      |`JSON int`|`1`|
 
-### File Scan Task Serialization
-
-File scan task is serialized as a JSON object according to the following table.
-
-| Metadata field       |JSON representation|Example|
-|--------------------------|--- |--- |
-| **`schema`**          |`JSON object`|`See above, read schemas instead`|
-| **`spec`**            |`JSON object`|`See above, read partition specs instead`|
-| **`data-file`**       |`JSON object`|`See above, read content file instead`|
-| **`delete-files`**    |`JSON list of objects`|`See above, read content file instead`|
-| **`residual-filter`** |`JSON object: residual filter expression`|`{"type":"eq","term":"id","value":1}`|
+### Task Serialization
+
+There could be different implementations of tasks, 
+e.g., `BaseFileScanTask` and `StaticDataTask` in Java.
+A enum `task-type` field is needed to distinguish different task types.
+
+| Metadata field  | JSON representation | Example                                                                                        |
+|-----------------|---------------------|------------------------------------------------------------------------------------------------|
+| **`task-type`** | `JSON string`       | `file-scan-task`, `data-task`. Absence of this field should be interpreted as `file-scan-task` |
+
+`file-scan-task` represents a scan task with a data file and maybe delete files.
+It is serialized according to the following table (in addition to the `task-type` field).
+
+| Metadata field         |JSON representation|Example|
+|------------------------|--- |--- |
+| **`schema`**           |`JSON object`|`See above, read schemas instead`|
```

Comment:
I just realized that I'm not clear on what the example `See above, read schemas instead` is trying to tell the reader. Should this maybe say `See "schemas" section above`?